core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -314,7 +314,7 @@ private int maybeResetInputAndSeekToEndIntermediateTopicOffsets(final Map consum
 
         try (final KafkaConsumer<byte[], byte[]> client = new KafkaConsumer<>(config, new ByteArrayDeserializer(), new ByteArrayDeserializer())) {
             client.subscribe(topicsToSubscribe);
-            client.poll(1);
+            client.poll(java.time.Duration.ofMillis(1));
```

Comment:
@mjsax I think you authored the change where this line was introduced. Do you know what's the reasoning for poll(1)? Did it just used the fact that it updated the metadata or were there any other reasons? I'm not sure that my change is correct here and I don't have the context.

Reply:
Yes, we do this call to update the metadata. The change seems reasonable to me.

Reply:
Are you sure about  this? `poll(1)` would previously have guaranteed that the consumer joins the group. That is no longer the case with the new poll, so the call to `assignment` below is likely to return an empty set.

Reply:
Because `StreamsResetter` must be the only member in the group (we check on startup that the group is empty), we can check if the assignment contains all offsets? Or we could switch to manual assignment?

Reply:
Yeah, I think manual assignment is a better option here. For one thing, it guarantees that there won't be another member in the group when you receive the assignment. It also gets around the awkward use of poll() since you do not actually want to fetch data. The basic approach would be to use the `partitionsFor` API to find the partitions for the subscribed topics and then assign them directly with `assign`.

Reply:
Ok, I'll try to refactor my code this way.